samu.c

```diff
@@ -90,6 +91,50 @@ jobsflag(const char *flag)
 	buildopts.maxjobs = num > 0 ? num : -1;
 }
 
+static void
+jobserverflags(const char *flag)
+{
+	int rfd, wfd;
+
+	if (!flag)
+		return;
+	if (sscanf(flag, "%d,%d", &rfd, &wfd) == 2) {
+		/* prepare error message */
+		errno = EBADF;
+	} else if (strncmp(flag, "fifo:", 5) == 0) {
+		rfd = open(flag + 5, O_RDONLY);
+		wfd = open(flag + 5, O_WRONLY);
+	} else {
+		fatal("invalid jobserver parameter");
+	}
+
+	buildopts.gmakepipe[0] = rfd;
+	buildopts.gmakepipe[1] = wfd;
+	warn("using GNU Make jobserver");
+}
+
+static void
+parsegmakeflags(char *env) {
+	char *arg;
+
+	if (!env)
+		return;
+	env = xmemdup(env, strlen(env) + 1);
+
+	arg = strtok(env, " ");
+	while (arg) {
+		if (arg[0] && arg[0] != '-' && strchr(arg, 'n')) {
+			buildopts.dryrun = true;
+		} else if (strncmp(arg, "-j", 2) == 0) {
+			jobsflag(arg + 2);
+		} else if (strncmp(arg, "--jobserver-auth=", 17) == 0 || strncmp(arg, "--jobserver-fds=", 16) == 0) {
```

Comment:
I don't think we need to support `--jobserver-fds`. It's not documented, so I assume it's an old name for the flag.

Reply:
It's an old name for it (before GNU Make 4.2, I think). As far as I can tell, LTS distros don't use GNU Make 4.1.x anymore, so it can be safely dropped.